indico/modules/events/papers/lists.py

```diff
@@ -0,0 +1,140 @@
+        ])
+        self.list_config = self._get_config()
+
+    def _get_static_columns(self, ids):
+        """
+        Retrieve information needed for the header of the static columns.
+
+        :return: a list of {'id': ..., 'caption': ...} dicts
+        """
+        return [{'id': id_, 'caption': self.static_items[id_]['title']} for id_ in self.static_items if id_ in ids]
+
+    def _build_query(self):
+        return (Contribution.query.with_parent(self.event)
+                .order_by(Contribution.friendly_id)
+                .options(subqueryload('paper_last_revision')))
+
+    def _filter_list_entries(self, query, filters):
+        if not filters.get('items'):
+            return query
+        criteria = []
+        if 'state' in filters['items']:
+            filtered_states = filters['items']['state']
+            state_criteria = []
+            for filter_state in filtered_states:
+                if filter_state is None:
+                    state_criteria.append(~Contribution.paper_last_revision.has())
+                else:
+                    state_criteria.append(Contribution.paper_last_revision
+                                          .has(PaperRevision.state == int(filter_state)))
+            if state_criteria:
+                criteria.append(db.or_(*state_criteria))
+
+        filter_cols = {'track': Contribution.track_id,
+                       'session': Contribution.session_id,
+                       'type': Contribution.type_id}
+        for key, column in filter_cols.iteritems():
+            ids = set(filters['items'].get(key, ()))
+            if not ids:
+                continue
+            column_criteria = []
+            if None in ids:
+                column_criteria.append(column.is_(None))
+            if ids - {None}:
+                column_criteria.append(column.in_(ids - {None}))
+            criteria.append(db.or_(*column_criteria))
+
+        return query.filter(*criteria)
+
+    def get_list_kwargs(self):
+        list_config = self._get_config()
+        contributions_query = self._build_query()
+        total_entries = contributions_query.count()
+        contributions = self._filter_list_entries(contributions_query, self.list_config['filters']).all()
+        selected_entry = request.args.get('selected')
+        selected_entry = int(selected_entry) if selected_entry else None
+        static_item_ids, dynamic_item_ids = self._split_item_ids(list_config['items'], 'static')
+        static_columns = self._get_static_columns(static_item_ids)
+        return {'contribs': contributions, 'total_entries': total_entries, 'selected_entry': selected_entry,
+                'static_columns': static_columns}
+
+    def render_list(self):
+        """Render the contribution list template components.
+
+        :return: dict containing the list's entries, the fragment of
+                 displayed entries and whether the contrib passed is displayed
```

Comment:
*"...whether the contrib**ution**..."*, it's docstring, no need for abbreviations!